pkglistgen/cli.py

```diff
@@ -95,6 +102,7 @@ def solve_project(project, scope: str):
                 self.tool.dry_run = self.options.dry
                 return self.tool.update_and_solve_target(api, target_project, target_config, main_repo,
                                                          git_url=opts.git_url, project=project, scope=scope,
+                                                         use_product_composer=(opts.engine == Engine.product_composer.name),
```

Comment:
I wonder if we should simply pass the engine rather than converting it to a fixed `use_product_composer` boolean. Doing comparisons against `Engine.product_composer` or `Engine.legacy` is more clear IMHO.